specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/CommonDefinitions.json

```diff
@@ -3586,6 +3590,157 @@
         }
       }
     },
+    "FunctionsDeployment": {
+      "description": "Configuration section for Azure Function app deployment.",
+      "type": "object",
+      "properties": {
+        "storage": {
+          "description": "Storage for deployed package used by the Azure Function app.",
+          "type": "object",
+          "properties": {
+            "type": {
+              "description": "Property to select Azure Storage Type. Available options: blobContainer.",
+              "type": "string",
+              "enum": [
+                "blobContainer"
+              ],
+              "x-ms-enum": {
+                "name": "type",
+                "modelAsString": true
+              }
+            },
+            "value": {
+              "description": "Property to set the URL for the selected Azure Storage Type. Example: For blobContainer, value will be like https://<storageAccountName>.blob.core.windows.net/<containerName>.",
+              "type": "string",
+              "format": "uri"
+            },
+            "authentication": {
+              "description": "Authentication method to access the storage for deployment.",
+              "type": "object",
+              "properties": {
+                "type": {
+                  "description": "Property to select Authentication Type to access the selected storage. Available options: SystemAssignedIdentity, UserAssignedIdentity, StorageAccountConnectionString.",
+                  "type": "string",
+                  "enum": [
+                    "SystemAssignedIdentity",
+                    "UserAssignedIdentity",
+                    "StorageAccountConnectionString"
+                  ],
+                  "x-ms-enum": {
+                    "name": "type",
+                    "modelAsString": true
+                  }
+                },
+                "userAssignedIdentityResourceId": {
+                  "description": "Use this property for UserAssignedIdentity. Set the resource ID of the identity. Do not set a value for this property when using other authentication type.",
+                  "type": "string"
+                },
+                "storageAccountConnectionStringName": {
```

Comment:
>storageAccountConnectionStringName [](http://example.com/codeflow?start=17&length=34)

Does this need to be marked as x-ms-secret ?

Reply:
@raosuhas  -- The `storageAccountConnectionStringName` refers to the name of the app setting that contains the actual connection string (a secret). Therefore, I think we do not need to mark it as `x-ms-secret`.

In the example below, `SomeAppSettingName` is the name of the app setting that contains the connection string. 

```json
"functionAppConfig": {
    "deployment": {
      "storage": {
        "type": "blobContainer",
        "value": "https://storageAccountName.blob.core.windows.net/containername",
        "authentication": {
          "type": "StorageAccountAccessKey",
          "storageAccountConnectionStringName": "SomeAppSettingName"
        }
      }
    }
// ...
```